docs/dynamic-component-loading.md

```diff
@@ -0,0 +1,48 @@
+# Dynamic component loading
+
+## Problem statement
+.NET Core runtime has only limited support for dynamically loading assemblies which have additional dependencies or possibly collide with the app in any way. Out of the box only these scenarios really work:
+* Assemblies which have no additional dependencies other than those find in the app itself (`Assembly.Load` and `Assembly.LoadFrom`)
+* Assemblies which have additional dependencies in the same folder and which don't collide with anything in the app itself (`Assembly.LoadFrom`)
```

Comment:
In general this does not work for the dynamic case.  The only way dependencies are found is if they are described in the main *.deps.json.

Reply:
I think this point is trying to capture the dependencies specified via `--additional-deps`

Reply:
This section describes current behavior, basically trying to describe why a new thing is needed. 

`Assembly.LoadFrom` currently implements a mechanism to resolve dependencies which are in the same folder as the loaded assembly. It's the closest thing we have to "Dynamically load a component". It doesn't look at `.deps.json` in any way, and it loads everything into the Default context which makes it pretty hard to use.